includes/wpml/class-language-details.php

```diff
@@ -0,0 +1,145 @@
+<?php
+/**
+ * File containing the \Sensei\WPML\Language_Details class.
+ *
+ * @package sensei
+ */
+
+namespace Sensei\WPML;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Class Language_Details
+ *
+ * Compatibility code with WPML.
+ *
+ * @since $$next-version$$
+ *
+ * @internal
+ */
+class Language_Details {
+	/**
+	 * Init hooks.
+	 */
+	public function init() {
+		add_action( 'sensei_course_structure_lesson_created', array( $this, 'set_language_details_when_lesson_created' ), 10, 2 );
+		add_action( 'sensei_course_structure_quiz_created', array( $this, 'set_language_details_when_quiz_created' ), 10, 2 );
```

Comment:
Not critical, but something to think about.

WDYT about having just a single hook for quiz creation? And maybe have a `$context` variable to identify where it's being called?

Also, I noticed `sensei_lesson_quiz_created` is being called based on the `save_post` hook. So it wouldn't apply for the 3 cases we have here (I didn't dig into the method to understand everything that happens there)?

If not, based on the name of the hook, wouldn't a dev expect that `sensei_lesson_quiz_created` is always called?

Reply:
> WDYT about having just a single hook for quiz creation? And maybe have a $context variable to identify where it's being called?

Do you mean to fire the same action (~~`sensei_lesson_quiz_created`~~ `sensei_lesson_create_quiz`), but instead of a difference in name put that information into context argument?
Makes sense, but ...
It contradicts our naming conventions (P6rkRX-4oA-p2) a bit (look at the second part of the name - screen or functional area — do you have any ideas on how to apply it to our case?), but at the same time I just found my naming also has a mistake according to conventions (the ending should be `_create_quiz`).

Reply:
> Do you mean to fire the same action (sensei_lesson_quiz_created sensei_lesson_create_quiz), but instead of a difference in name put that information into context argument?

Yes, but I think without the `lesson` word. Maybe something like `sensei_create_quiz`?

> It contradicts our naming conventions (P6rkRX-4oA-p2) a bit (look at the second part of the name - screen or functional area — do you have any ideas on how to apply it to our case?), but at the same time I just found my naming also has a mistake according to conventions (the ending should be _create_quiz).

Hm! Good point! I'm not sure how to fit this hook into this convention. 🤔 I'm not totally sure if `quiz` should be our `<element>` or our `<screen_or_functional_area>`.

Maybe it should be both? `sensei_quiz_create_quiz`. That's just a little weird.

I'd say to open an exception for this type of cases. But if we agree on it, what is the type of case to document this exception? Maybe when the `<screen_or_functional_area>` and `<element>` are the same we should use only the second one? Or maybe we should always make the `<element>` optional and use `sensei_quiz_create` in this case?

The other option that I can think of that would fit into this convention is the `sensei_lesson_create_quiz` that you suggested. Being:
- `<plugin_name>`- sensei
- `<screen_or_functional_area>` - lesson
- `<optional_verb>` - create
- `<element>` - quiz
But it could become wrong in the future if we decide to create quizzes not tied to lessons (a final quiz in a course, a quiz in a post, or anything else).

Reply:
I think we can use `sensei_quiz_create` then:

- `<plugin_name>` - sensei
- `<screen_or_functional_area>` - quiz (I use the analogy of a resource from REST API)
- `<optional_verb>` - create
- `<element>` is optional, we don't need it in this case.

I'm making changes applying this rule. Let me know if you're not agree.

Reply:
Yep! Looks good to me! Thanks!